pystac_client/warnings.py

```diff
@@ -0,0 +1,73 @@
+from contextlib import contextmanager
+import warnings
+
+
+FALLBACK_MSG = "Falling back to pystac. This might be slow."
+
+
+class PystacClientWarning(UserWarning):
+    """Base warning class"""
+
+    ...
+
+
+class NoConformsTo(PystacClientWarning):
+    """Inform user when client does not have "conformsTo" set"""
+
+    ...
+
+
+class DoesNotConformTo(PystacClientWarning):
+    """Inform user when client does not conform to extension"""
+
+    ...
+
+
+class MissingLink(PystacClientWarning):
+    """Inform user when link is properly implemented"""
+
+    ...
+
+
+class FallbackToPystac(PystacClientWarning):
+    """Inform user when falling back to pystac implementation"""
+
+    ...
+
+
+@contextmanager
+def strict():  # type: ignore
+    """Context manager for raising all pystac-client warnings as errors
+
+    For more fine-grained control or to set filter warnings in the whole
+    python session, use the ``warnings`` module directly.
+
+    Examples:
+
+    >>> import warnings
+    ... warnings.filterwarnings("error", category=FallbackToPystac)
+    """
+    warnings.filterwarnings("error", category=PystacClientWarning)
+    try:
+        yield
+    finally:
+        warnings.resetwarnings()
+
+
+@contextmanager
+def lax():  # type: ignore
```

Comment:
```suggestion
def ignore():  # type: ignore
```

`lax`, while a good term, feels one step removed from the core purpose of the function (to ignore stuff).

Reply:
yes I did not expect that one to stick (pun intended).